File: 백준/Silver/14248. 점프 점프/점프 점프.java
import java.io.*;
import java.util.*;


public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int N = Integer.parseInt(br.readLine());
        int[] lst = new int[N];
        int[] visited = new int[N];
        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < N; i++) {
            lst[i] = Integer.parseInt(st.nextToken());
            visited[i] = 0;}
        st = new StringTokenizer(br.readLine());
        int S = Integer.parseInt(st.nextToken())-1;
        Queue<Integer> q = new LinkedList<>();
        q.add(S);
        while (q.size() > 0) {
            int x = q.poll();
            visited[x] = 1;
            if (x - lst[x] >= 0 && visited[x - lst[x]] == 0) {
                q.add(x - lst[x]);
            }
            if (x + lst[x] < N && visited[x + lst[x]] == 0) {
                q.add(x + lst[x]);
            }
        }
        int ans = 0;
        for (int i = 0; i < N; i++) {ans+=visited[i];}
        bw.write(Integer.toString(ans));
        bw.close();
    }
}

File: 백준/Silver/14248. 점프 점프/점프 점프.py
import sys
input = sys.stdin.readline
from collections import deque

def Main():
    N=int(input())
    lst = list(map(int,input().split()))
    visited = [0]*N
    S = int(input())-1
    ans = 0
    q = deque()
    q.append(S)
    while q:
        x= q.popleft()
        visited[x] = 1
        if x-lst[x] >= 0 and visited[x-lst[x]] ==0:
            q.append(x-lst[x])
        if x+lst[x] < N and visited[x+lst[x]]==0:
            q.append(x+lst[x])
    print(sum(visited))
Main()